src/eda.py

```diff
@@ -0,0 +1,82 @@
+# author: Linh Giang Nguyen
+# date: 2021-11-28
+
+"""Creates eda plots for the pre-processed training data from the National Collision Database (NCDB) 2017 data (from https://open.canada.ca/data/en/dataset/1eb9eba7-71d1-4b30-9fb1-30cbdab7e63a/resource/01426d41-529c-443f-a901-6bc2f94f3d73).
+Saves the plots as png files.
+
+Usage: src/eda_ncdb.py --train=<train> --out_dir=<out_dir>
+
+Options:
+--train=<train>     Path (including filename) to training data (which needs to be saved as a feather file)
+--out_dir=<out_dir> Path to directory where the plots should be saved
+"""
+
+from docopt import docopt
+import pandas as pd
+import altair as alt
+
+alt.data_transformers.enable('data_server')
+alt.renderers.enable('mimetype');
+
+opt = docopt(__doc__)
+
+def create_and_save_chart(df, col, title, save_path):
+    """
+    Returns distribution plots of the feature of interest from the dataframe `nc_train_nan`.
+    The plot is faceted by class (no-fatality: blue, fatality: orange).
+
+    Parameters
+    ----------
+    col : str
+        the feature of interest
+    title : str
+        the label for the feature in the returned plot
+
+    Returns
+    -------
+    out: altair.vegalite.v4.api.FacetChart 
+       the faceted distribution plots 
+    
+    Examples
+    --------
+    >>> create_chart('C_WTHR', 'Weather condition')
+    """
+    
+    if col == 'P_AGE':
+        chart = alt.Chart(df).mark_line().encode(
+                    x=alt.X('P_AGE:Q', title=title),
```

Comment:
It will also affect Doris' work as she will have to use StandardScaler on the age column instead of OHE

Reply:
I need to specify the type here just in order for the chart to work (because of the mark_line() I think). But in the back end, I think you can keep the type as is (similar to other columns). What do you think?

Reply:
Yah I think age makes sense as a categorical since it is discrete